sdk/state/integrationtests/state_test.go

```diff
@@ -42,37 +42,46 @@ func TestOpenUpdatesUncoordinatedClose(t *testing.T) {
 
 	// Open
 	t.Log("Open...")
+	// I signs txClose
 	open, err := initiatorChannel.ProposeOpen(state.OpenParams{Asset: asset, AssetLimit: assetLimit})
 	require.NoError(t, err)
-	for {
```

Comment:
changed this to do the steps explicitly. I found it difficult to understand how many times each channel calls confirmOpen with the `for loop`

Reply:
Do you think this suggests the approach we are taking where we call a single function to do confirmation is a mistake? Maybe we do need separate functions for each step of we want the caller to know what's happening each each step. Originally the goal was to abstract that so a caller wouldn't need to know.

Reply:
🤔  it does make it clearer if we split it. But I agree that the caller shouldn't need to know which of the transactions are confirmed or not confirmed at a point in time. This change was more for me/us while we're developing to make it clearer.

In practice I think a user calling `ConfirmOpen` until the authorized flag turns true, is a better interface.

Otherwise it could become something like `ConfirmOpen/ReConfirmOpen/ConfirmFormation` or something similarly ugly. Also if we split it then it gets complicated in what if someone calls `ReConfirmOpen` before `ConfirmOpen` or another out of order step etc.

Reply:
Got it. This probably makes the test a bit more predictable too.

Reply:
We don't have to do this in this PR, but now that this is changed to be more explicit we could add some assertions to this test after each call to check the right number of signers are present, or the right signers are present.

Reply:
added :)
https://github.com/stellar/experimental-payment-channels/pull/125/commits/01c3803648356e523978d5f15d00efc8e316384e